### PR TITLE
Integrate IPsec certificate of atsec cases into openQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/110734

Verify run:
               https://openqa.suse.de/tests/8751002# (x86_64)
               https://openqa.suse.de/tests/8764147# (aarch64)
               https://openqa.suse.de/tests/8764274# (s390x)

### CODE REVIEW

File: tests/security/cc/ipsec/ipsec_certificate.pm [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                $record_message = "The $case_name test result is NOT expected";
+                $self->result('fail');
+            }
+            else {
+                if ($case_name eq 'rsa768') {
+                    $result = 'softfail';
+                    $record_message = "$case_name pass, as ATSec document says, it needs more analysis";
+
+                }
+                # When the ipsec up successes, we need to check if the connection is created
```
Comment (rfan1) [inline tests/security/cc/ipsec/ipsec_certificate.pm:79]: Succeed?

File: tests/security/cc/ipsec/ipsec_server.pm [hunk 2]
```diff
@@ -41,16 +39,28 @@ sub run {
...
+
+    mutex_create('READY_FOR_IPSEC_CIPHERS');
+
+    # Change the configure file for testing ipsec ciphers
+    my $new_algorithms = ',aes256ctr-sha256-modp2048,aes128ctr-sha256-modp6144,aes256ccm8-prfsha256-modp3072';
+    assert_script_run("sed -i '/ proposals = / s/\$/$new_algorithms/' /etc/swanctl/conf.d/ikev2suse.conf");
+    script_run('cat /etc/swanctl/conf.d/ikev2suse.conf');
+
+    # restart the strongswan service
+    assert_script_run('systemctl restart strongswan');
```
Comment (rfan1) [inline tests/security/cc/ipsec/ipsec_server.pm:55]: I suggest using existing 'systemctl' function, but not a big deal

File: tests/security/cc/ipsec/ipsec_server.pm [hunk 3]
```diff
@@ -41,16 +39,28 @@ sub run {
...
+    mutex_wait('WEAK_IPSEC_CIPHERS_DONE', (keys %$children)[0]);
+
+    mutex_create('READY_FOR_IPSEC_CIPHERS');
+
+    # Change the configure file for testing ipsec ciphers
+    my $new_algorithms = ',aes256ctr-sha256-modp2048,aes128ctr-sha256-modp6144,aes256ccm8-prfsha256-modp3072';
+    assert_script_run("sed -i '/ proposals = / s/\$/$new_algorithms/' /etc/swanctl/conf.d/ikev2suse.conf");
+    script_run('cat /etc/swanctl/conf.d/ikev2suse.conf');
+
+    # restart the strongswan service
```
Comment (lilyeyes) [inline tests/security/cc/ipsec/ipsec_server.pm:54]: s/r/R/

File: tests/security/cc/ipsec/weak_ipsec_ciphers.pm [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # Copyright 2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 #
-# Summary: Run CC 'ipsec' client case
+# Summary: Run ATSec ' Weak IPsec ciphers' case
```
Comment (lilyeyes) [inline tests/security/cc/ipsec/weak_ipsec_ciphers.pm:6]: s/ Weak/Weak/

**Review by wangqianzou — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Pull request merged**